Sources/TokamakCore/Views/Gestures/_GestureView.swift

```
@@ -0,0 +1,134 @@
+    
+    let mask: GestureMask
+    let priority: _GesturePriority
+    public let content: Content
+    
+    var minimumDuration: Double? {
+        guard let longPressGesture = gesture as? LongPressGesture else {
+            return nil
+        }
+        return longPressGesture.minimumDuration
+    }
+    
+    public init(
+        gesture: G,
+        mask: GestureMask,
+        priority: _GesturePriority = .standard,
+        content: Content
+    ) {
+        self._gesture = State(wrappedValue: gesture)
+        self.mask = mask
+        self.priority = priority
+        self.content = content
+    }
+    
+    public func onPhaseChange(_ phase: _GesturePhase, eventId id: String? = nil) {
+        guard isEnabled, let currentEventId = eventId ?? id else { return }
+        
+        let value = GestureValue(gestureId: gestureId, mask: mask, priority: priority)
+
+        switch phase {
+        case .began:
+            startDelay()
+            eventId = id
+            gestureListener.registerStart(value, for: currentEventId)
+        case .cancelled, .ended:
+            eventId = nil
+        default:
+            break
+        }
+        
+        guard gestureListener.canProcessGesture(value, for: currentEventId) else {
+            // Event being processed by another gestures
+            return
+        }
+        
+        if gesture._onPhaseChange(phase) {
+            gestureListener.recognizeGesture(value, for: currentEventId)
+        }
+    }
+    
+    private func startDelay() {
+        guard let minimumDuration else { return }
+        Task {
+            do {
+                try await Task.sleep(for: .seconds(minimumDuration))
+                if let eventId {
+                    await MainActor.run {
+                        onPhaseChange(.changed(location: nil), eventId: eventId)
+                    }
+                }
+            } catch {
+                //TODO: What do we do with this error?
+                print(error)
```

I think it would only fail if the task is cancelled before the `Task.sleep` finished. So it might be ok to ignore the error instead of logging it?
```suggestion
```

---

fixed